src/host-config.ts

```diff
@@ -45,41 +52,83 @@ const memoizedRegex = _.memoize(
 	(proxyField) => new RegExp('^\\s*' + proxyField + '\\s*=\\s*([^;\\s]*)\\s*;'),
 );
 
-async function readProxy(): Promise<ProxyConfig | undefined> {
-	const conf: ProxyConfig = {};
-	let redsocksConf: string;
+async function readProxy(): Promise<
+	[Nullable<ProxyConfig>, Nullable<DnsConfig>]
+> {
+	let rawRedsocksConf: string;
 	try {
-		redsocksConf = await readFromBoot(redsocksConfPath, 'utf-8');
+		rawRedsocksConf = await readFromBoot(redsocksConfPath, 'utf-8');
 	} catch (e: unknown) {
 		if (!isENOENT(e)) {
 			throw e;
 		}
-		return;
+		return [null, null];
 	}
-	const lines = redsocksConf.split('\n');
 
-	for (const line of lines) {
-		for (const proxyField of proxyFields) {
-			let match: string[] | null = null;
-			if (isAuthField(proxyField)) {
-				match = line.match(memoizedAuthRegex(proxyField));
-			} else {
-				match = line.match(memoizedRegex(proxyField));
+	const redsocksConfBlocks: [Nullable<ProxyConfig>, Nullable<DnsConfig>] = [
+		null,
+		null,
+	];
+
+	// Each configuration block should be separated by two newlines
+	const confBlocks = rawRedsocksConf.split('\n\n');
+
+	// Handle redsocks block
+	const proxyConf: ProxyConfig = {};
+	const redsocksBlock = confBlocks.find((block) =>
+		block.includes('redsocks {'),
+	);
+	if (redsocksBlock != null) {
+		const proxyLines = redsocksBlock.split('\n');
+		for (const line of proxyLines) {
+			for (const proxyField of proxyFields) {
+				let match: string[] | null = null;
+				if (isAuthField(proxyField)) {
+					match = line.match(memoizedAuthRegex(proxyField));
+				} else {
+					match = line.match(memoizedRegex(proxyField));
+				}
+
+				if (match != null) {
+					proxyConf[proxyField] = match[1];
+				}
 			}
+		}
+		redsocksConfBlocks[0] = proxyConf;
+	}
 
-			if (match != null) {
-				conf[proxyField] = match[1];
+	// Handle dns block
+	const dnsConf: DnsConfig = {};
+	const dnsBlock = confBlocks.find((block) => block.includes('dnsu2t {'));
+	if (dnsBlock != null) {
+		const dnsLines = dnsBlock.split('\n');
+		for (const line of dnsLines) {
+			for (const dnsField of dnsFields) {
+				const match = line.match(memoizedRegex(dnsField));
+				if (match != null) {
+					// Transform snakecase dnsField to camelCase
+					const dnsFieldCamelcase = dnsField.replace(
+						/* eslint-disable @typescript-eslint/no-shadow */
+						// This is necessary because ESLint mistakes _ here
```

Comment:
The argument doesn't need to be called `_`, it just has to start with `_`, e.g. `_ignore`